### PR TITLE
Add generator meta tag for Embed Optimizer

### PR DESCRIPTION
**Pull request by westonruter**
This matches the `meta[name=generator]` added by WebP Uploads and Dominant Color Images.

### CODE REVIEW

File: plugins/embed-optimizer/hooks.php [hunk 1]
```diff
@@ -153,3 +153,19 @@ function embed_optimizer_trigger_error( string $function_name, string $message,
 	}
 	wp_trigger_error( $function_name, $message, $error_level );
 }
+
+/**
+ * Displays the HTML generator tag for the WebP Uploads plugin.
```
Comment (adamsilverstein) [inline plugins/embed-optimizer/hooks.php]:
```suggestion
 * Displays the HTML generator tag for the Embed Optimizer plugin.
```

File: plugins/embed-optimizer/hooks.php [hunk 2]
```diff
@@ -153,3 +153,19 @@ function embed_optimizer_trigger_error( string $function_name, string $message,
...
+ * Displays the HTML generator tag for the Embed Optimizer plugin.
+ *
+ * See {@see 'wp_head'}.
+ *
+ * @since 0.1.0
+ */
+function embed_optimizer_render_generator() {
+	if (
+		defined( 'EMBED_OPTIMIZER_VERSION' )
+	) {
```
Comment (westonruter) [inline plugins/embed-optimizer/hooks.php:167]: This condition seems pointless if we always define the constant. But it is present in the other plugins.

File: tests/plugins/embed-optimizer/embed-optimizer-test.php [hunk 3]
```diff
@@ -156,4 +156,17 @@ function ( array $attrs, string $javascript ) {
 		$this->assertStringStartsWith( '<script type="module" nonce="abc123">', $script );
 		$this->assertStringEndsWith( '</script>', $script );
 	}
+
+	/**
+	 * Test printing the meta generator tag.
+	 *
+	 * @covers ::embed_optimizer_render_generator
+	 */
+	public function test_embed_optimizer_render_generator() {
```
Comment (adamsilverstein) [inline tests/plugins/embed-optimizer/embed-optimizer-test.php:165]: Nice, very thorough!

File: plugins/embed-optimizer/hooks.php (hunk 2 above)
Comment (adamsilverstein) [reply plugins/embed-optimizer/hooks.php:167]: 🤷🏼 very defensive coding

**Review by adamsilverstein — APPROVED** [review]
Helpful!

File: plugins/embed-optimizer/hooks.php (hunk 2 above)
Comment (westonruter) [reply plugins/embed-optimizer/hooks.php:167]: Maybe we should also check if `esc_attr()` exists and wrap it in a `try`/`catch` block then 😄

**Pull request merged**